src/generators/angular/AngularServicesMethodGenerator.ts

```diff
@@ -22,14 +24,14 @@ interface IParameterType {
 }
 
 export class AngularServicesMethodGenerator {
-    constructor(private uriBuilder: UriBuilder) {}
+    constructor(private uriBuilder: UriBuilder, private settings: IOptions) {}
 
     public getMethodsCodeStructures(methodModels: IMethodModel[]): OptionalKind<MethodDeclarationStructure>[] {
         return methodModels.map(
             (methodModel): OptionalKind<MethodDeclarationStructure> => ({
                 scope: Scope.Public,
                 name: methodModel.name,
-                parameters: methodModel.parameters.map((p) => this.getParameterStatement(p)),
+                parameters: this.getMethodParameters(methodModel).map((p) => this.getParameterStatement(p)),
```

Comment:
I guess we should separate our parameters from swagger and specific generation parameters (e.g. request options)

So I suggest solution like:

1) Method getMethodParameters(methodModel) that would return ParameterDeclarationStructure[]
2) getMethodParameters would map our methodModel.parameters
and adds option parameter as needed
Something like
```ts
public getMethodParameters(model) {
     const statements = model.parameters.map(x => this.getParameterStatement(x));

     if (this.options.withRequestOptions) {
          statements.push(this.getRequestOptionsParameter());
     }

     return statements;
}
```

that way we would not depend on the structure of our swagger based parameters.
and would not mutate model.parameters (which is external data for this service)